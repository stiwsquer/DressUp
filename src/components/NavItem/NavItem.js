import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";
export default function NavItem({
  linkTo,
  iconClass,
  text,
}) {
  return (
    <Link to={linkTo}>
      <i className={iconClass}></i>
      {iconClass !== undefined && iconClass !== "" && " "}
      {text}
    </Link>
  );
}
