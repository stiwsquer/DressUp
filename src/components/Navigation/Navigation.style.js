import styled, { keyframes } from "styled-components";
import { flex } from "../../GlobalStyles";

const logoAnimation = keyframes`
  0% {
    transform: translateX(-50%) translateY(-200%);
  }
  100% {
    transform: translateX(-50%) translateY(35%);
  }
  
`;

export const NavLinks = styled.nav`
  min-height: 15vh;
  position: relative;
  width: 95%;
  margin: auto;
  ${flex({ justify: "space-between", align: "center" })};
  li,
  .cart {
    font-size: 1.6rem;
    margin: 2rem 1rem;
    text-align: center;
    white-space: nowrap;
    list-style: none;
  }

  li a,
  .cart a {
    transition: all 0.4s ease-in-out;
    transform: scale(1);
  }
  li a:hover,
  .cart a:hover {
    transform: scale(1.2);
    padding: 1rem;
    border-radius: 1.5rem;
    background: rgba(20, 20, 20, 1);
    color: white;
    font-weight: 900;
  }

  .logo {
    font-size: 6rem;
    position: absolute;
    left: 50%;
    bottom: 50%;
    transform: translateX(-50%) translateY(30%);
    transform: translateX(-50%) translateY(-200%);
    animation: ${logoAnimation} 1s ease-in-out 0.5s forwards;
  }

  @media screen and (max-width: 1000px) {
    .logo a {
      font-size: 2.5rem;
    }
  }
`;

export const Burger = styled.span`
  display: none;

  @media screen and (max-width: 1000px) {
    display: block;
    font-size: 2rem;
    font-size: 1.6rem;
    z-index: 2;
  }
`;
