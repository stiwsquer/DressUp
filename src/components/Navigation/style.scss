@import "../../styles/global.scss";
@keyframes logo-animation {
  0% {
    transform: translateX(-50%) translateY(-200%);
  }
  100% {
    transform: translateX(-50%) translateY(35%);
  }
}

.nav-links {
  min-height: 15vh;
  position: relative;
  width: 95%;
  margin: auto;
  @include flex(row, space-between, center);
  li,
  .cart {
    font-size: $nav-links;
    margin: 2rem 1rem;
    text-align: center;
    white-space: nowrap;
    list-style: none;
    a {
      transition: all 0.4s ease-in-out;
      transform: scale(1);
    }
    a:hover {
      transform: scale(1.2);
      padding: 1rem;
      border-radius: 1.5rem;
      background: rgba($color: #202020, $alpha: 1);
      color: white;
      font-weight: 900;
    }
  }
}
.burger,
.mobile-menu {
  display: none;
}

.logo {
  font-size: $header1;
  position: absolute;
  left: 50%;
  bottom: 50%;
  transform: translateX(-50%) translateY(30%);
  transform: translateX(-50%) translateY(-200%);
  a:hover {
    color: black;
  }
  animation: logo-animation 1s ease-in-out 0.5s forwards;
}

@media screen and (max-width: 1000px) {
  .logo {
    a {
      font-size: $header3;
    }
  }

  .burger {
    display: block;
    font-size: 2rem;
    font-size: $nav-links;
    z-index: 2;
  }
}
