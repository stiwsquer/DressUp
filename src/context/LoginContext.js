import React, { useState, createContext, useContext } from "react";
import useCookie from "../hooks/useCookie/useCookie";
import Cookies from "js-cookie";

const LoginContext = createContext();

export const useLoginContext = () => {
  return useContext(LoginContext);
};

export const LoginContextProvider = ({ children }) => {
  const [isUserLoggedIn, setIsUserLoggedIn] = useState(false);

  function updateUserStatus(isUserLoggedIn) {
    // if (
    //   Cookies.get("access_token") != undefined &&
    //   Cookies.get("refresh_token") != undefined
    // ) {
    setIsUserLoggedIn(isUserLoggedIn);
    // } else {
    // setIsUserLoggedIn(false);
    // }
  }

  return (
    <LoginContext.Provider value={[isUserLoggedIn, updateUserStatus]}>
      {children}
    </LoginContext.Provider>
  );
};
