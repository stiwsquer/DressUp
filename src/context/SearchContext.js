import React, { useState, createContext, useContext } from 'react';

const SearchContext = createContext();

export const useSearchContext = () => useContext(SearchContext);

export const SearchContextProvider = ({ children }) => {
  const [input, setInput] = useState('');

  const setSearchInput = (e) => {
    setInput(e.target.value);
  };
  return (
    <SearchContext.Provider value={[input, setSearchInput]}>
      {children}
    </SearchContext.Provider>
  );
};
