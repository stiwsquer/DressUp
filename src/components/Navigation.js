import React, { useContext } from "react";
import { BrowserRouter as Link } from "react-router-dom";
import NavItem from "./NavItem";
export default function Navigation(props) {
  return (
    <nav className="nav-links">
      <ul className="signIn-register-search">
        <li>
          <NavItem
            linkTo="/signIn"
            iconClass="fas fa-sign-in-alt"
            text="Sign in"
          />
        </li>
        <li className="vertical-line">|</li>
        <li>
          <NavItem linkTo="/register" text="Register" />
        </li>
        <li className="vertical-line">|</li>
        <li>
          <NavItem linkTo="/search" iconClass="fas fa-search" text="Search" />
        </li>
      </ul>

      <ul className="logo-cart">
        <li>
          <h1 className="logo">
            <NavItem linkTo="/" text="DressUp" />
          </h1>
        </li>
        <li>
          <NavItem linkTo="#" iconClass="fas fa-shopping-cart" text="Cart" />
        </li>
      </ul>
    </nav>
  );
}
