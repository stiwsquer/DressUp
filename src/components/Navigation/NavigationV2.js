import React, { useLayoutEffect, useState, useContext, useRef } from "react";
import "./style.scss";
import NavItem from "../NavItem/NavItem";
import OutsideAlerter from "../OutsideAlerter/OutsideAlerter";
import SearchBox from "../SearchBox/SearchBox";
import SignInRegisterSearch from "../SignInRegisterSearch/SignInRegisterSearch";

export default React.memo(function Navigation() {
  const [showMenu, setShowMenu] = useState(false);
  const [showSearch, setShowSerch] = useState(false);
  const [width, height] = useWindowSize([0, 0]);
  const searchRef = useRef();

  const handleShowMenu = () => setShowMenu((prev) => !prev);

  let signInRegisterSearchClassName = "signIn-register-search";
  if (width < 1000 && !showMenu) {
    signInRegisterSearchClassName = "inactive-signIn-register-search";
  }

  return (
    <>
      <div>
        <nav className="nav-links">
          <span className="burger" onClick={handleShowMenu}>
            <i
              style={{ position: showMenu ? "fixed" : "" }}
              className={showMenu ? "fas fa-times" : "fas fa-bars"}
            ></i>
          </span>
          <SignInRegisterSearch
            searchRef={searchRef}
            nameOfTheClass={signInRegisterSearchClassName}
            setShowMenu={setShowMenu}
            showMenu={showMenu}
            setShowSerch={setShowSerch}
            showSearch={showSearch}
          />

          <h1 className="logo">
            <NavItem linkTo="/" text="DressUp" />
          </h1>

          <div className="cart">
            <NavItem linkTo="#" iconClass="fas fa-shopping-cart" text="Cart" />
          </div>
        </nav>
        {showSearch ? (
          <OutsideAlerter setShowElement={setShowSerch} searchRef={searchRef}>
            <SearchBox nameOfTheClass="search-box" />
          </OutsideAlerter>
        ) : null}
      </div>
    </>
  );
});

function useWindowSize() {
  const [size, setSize] = useState([0, 0]);
  useLayoutEffect(() => {
    function updateSize() {
      setSize([window.innerWidth, window.innerHeight]);
    }
    window.addEventListener("resize", updateSize);
    updateSize();
    return () => window.removeEventListener("resize", updateSize);
  }, []);
  return size;
}
