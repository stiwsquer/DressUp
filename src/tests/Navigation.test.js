import { BrowserRouter as Router } from "react-router-dom";
import Navigation from "../components/Navigation/NavigationV2";
import NavItem from "../components/NavItem/NavItem";
import { SearchContextProvider } from "../context/SearchContext";

it("renders Navigation component without crashing", () => {
  shallow(<Navigation />);
});

it("renders two <NavItem /> components", () => {
  const wrapper = shallow(<Navigation />);
  expect(wrapper.find(NavItem)).toHaveLength(2);
});

it("simulates opening mobile menu", () => {
  const wrapper = mount(
    <SearchContextProvider>
      <Router>
        <Navigation />
      </Router>
    </SearchContextProvider>
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("i").first().hasClass("fa-bars")).toEqual(true);

  wrapper.find("span").simulate("click");

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("i").first().hasClass("fa-times")).toEqual(true);
});
