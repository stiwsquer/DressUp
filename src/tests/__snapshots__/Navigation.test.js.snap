// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simulates opening mobile menu 1`] = `
<div>
  <nav
    className="nav-links"
  >
    <span
      className="burger"
      onClick={[Function]}
    >
      <i
        className="fas fa-bars"
        style={
          Object {
            "position": "",
          }
        }
      />
    </span>
    <ul
      className="signIn-register-search"
    >
      <li
        onClick={[Function]}
      >
        <a
          href="/signIn"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-in-alt"
          />
           
          Sign in
        </a>
      </li>
      <li
        className="vertical-line"
      >
        |
      </li>
      <li
        onClick={[Function]}
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          <i />
          Register
        </a>
      </li>
      <li
        className="vertical-line"
      >
        |
      </li>
      <li
        onClick={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="fas fa-search"
          />
           
          Search
        </a>
      </li>
    </ul>
    <h1
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i />
        DressUp
      </a>
    </h1>
    <div
      className="cart"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-shopping-cart"
        />
         
        Cart
      </a>
    </div>
  </nav>
</div>
`;

exports[`simulates opening mobile menu 2`] = `
<div>
  <nav
    className="nav-links"
  >
    <span
      className="burger"
      onClick={[Function]}
    >
      <i
        className="fas fa-times"
        style={
          Object {
            "position": "fixed",
          }
        }
      />
    </span>
    <ul
      className="signIn-register-search"
    >
      <li
        onClick={[Function]}
      >
        <a
          href="/signIn"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-in-alt"
          />
           
          Sign in
        </a>
      </li>
      <li
        className="vertical-line"
      >
        |
      </li>
      <li
        onClick={[Function]}
      >
        <a
          href="/register"
          onClick={[Function]}
        >
          <i />
          Register
        </a>
      </li>
      <li
        className="vertical-line"
      >
        |
      </li>
      <li
        onClick={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <i
            className="fas fa-search"
          />
           
          Search
        </a>
      </li>
    </ul>
    <h1
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i />
        DressUp
      </a>
    </h1>
    <div
      className="cart"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="fas fa-shopping-cart"
        />
         
        Cart
      </a>
    </div>
  </nav>
</div>
`;
